lib/logger-client.js

```diff
@@ -17,7 +17,7 @@ function createLoggerClient(config) {
   logger = createLogger(config.key, {
     payloadStructure: 'agent'
   , url: config.LOGDNA_URL
-  , hostname: config.hostname.replace(/[^a-zA-Z0-9]/g, '') // Satisfy hostname regex in the client
+  , hostname: config.hostname.replace(/[^a-zA-Z0-9.-]/g, '') // Satisfy hostname regex in the client
```

Comment:
Are we not putting any constraints on use of . and -? like start or end with a dash or not?

Reply:
should we just use the same regex as the client?
https://github.com/logdna/logger-node/blob/3bf0ecfb6cc417f97f7f974b11ac2cf7107810eb/lib/constants.js#L14

Reply:
TBH, I don't think `hostname` should have any constraints.  We don't want to have the client throw an error if it's too strict.  We're gonna have to look into the ingester to see if there was a reason why we're only accepting a certain format. For now, this change will still appease the `logger` while being a bit more flexible.

Reply:
works for me, and I agree that validating the hostname in general on the client side doesn't gain us much. the length is the only restriction on the ingestion side but it just gets trimmed.